src/components/lib/sl/sl.c

```diff
@@ -614,7 +617,7 @@ sl_sched_loop(void)
 			 * states of it's child threads) and normal notifications (mainly activations from
 			 * it's parent scheduler).
 			 */
-			pending = cos_sched_rcv(g->sched_rcv, RCV_ALL_PENDING, timeout,
+			pending = cos_sched_rcv(g->sched_rcv, rfl, timeout,
 						&rcvd, &tid, &blocked, &cycles, &thd_timeout);
 			if (!tid) goto pending_events;
 
```

Comment:
does the `g->event_head` not need a critical section?

Reply:
no it doesn't. this data structure is only accessed  by the scheduler thread in `sl_sched_loop` no where else. i'll remember to add a comment about that in my next PR.